org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java

```diff
@@ -398,12 +412,23 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
      * @param in input stream
      * @throws IOException
      */
-    static void writeToKnimeConsole(final InputStream in) throws IOException {
-        BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
-        String line;
-        while ((line = bis.readLine()) != null) {
-            KNIPGateway.log().info(line);
-        }
+    static void writeToKnimeConsole(final InputStream in) {
+
+        new Thread(
```

Comment:
> "redirectToKnimeConsole"[...] creates a thread which reads from the processes input stream to avoid it getting blocked and is called exactly twice per process invokation (one thread for stdout, one for stderr).
that's it.

The function should probably be called something like "redirectToKnimeConsole". It creates a thread which reads from the processes input stream to avoid it getting blocked and is called exactly twice per process invokation (one thread for stdout, one for stderr). I believe this one is correct? Or am I missing something?